scripts/tools/nrfconnect/nrfconnect_factory_data.schema

```diff
@@ -0,0 +1,99 @@
+{
+    "$id": "NrfConnect_Factory_Data_schema",
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "description": "A representation of all factory data used in Matter's NrfConnect device",
+    "type": "object",
+    "required": [
+        "sn",
+        "date",
+        "hw_ver",
+        "hw_ver_str",
+        "rd_uid",
+        "dac_cert",
+        "dac_key",
+        "pai_cert",
+        "cd",
+        "spake2_it",
+        "spake2_salt",
+        "spake2_verifier"
+    ],
+    "properties": {
+        "sn": {
+            "description": "Serial number of device",
+            "type": "string",
+            "maxLength": 32
+        },
+        "date": {
+            "description": "Manufacturing date according to ISO 8601 in notation YYYYMMDD",
+            "type": "string",
+            "minLength": 8,
+            "maxLength": 8,
+            "pattern": "[0-9]{4}[0-1]{1}[0-9]{1}[0-3]{1}[0-9]{1}"
+        },
+        "hw_ver": {
+            "description": "Hardware version - integer",
+            "type": "integer",
+            "minimum": 0,
+            "maximum": 65536
+        },
+        "hw_ver_str": {
+            "description": "A string representation of hardware version",
+            "type": "string",
+            "minLength": 1,
+            "maxLength": 64
+        },
+        "rd_uid": {
+            "description": "A randomly-generated 128-bit or longer octet string. Length has been expanded with 'hex:' prefix",
+            "type": "string",
+            "minLength": 20,
+            "maxLength": 36
+        },
+        "dac_cert": {
+            "description": "DAC certificate in hex-string format",
+            "type": "string"
+        },
+        "dac_key": {
+            "description": "DAC Private Key in hex-string format",
+            "type": "string"
+        },
+        "pai_cert": {
+            "description": "PAI certificate in hex-string format",
+            "type": "string"
+        },
+        "cd": {
```

Comment:
nit: cd does not look much verbose :) decl_cert?

Reply:
I tried to reduce the names of the keys written to flash, and I wanted to be coherent with names everywhere. I see what you're worried about, and I included descriptions in the schema and the script.